c++/src/capnp/compat/http-over-capnp.c++

```diff
@@ -306,8 +309,12 @@ public:
     });
 
     auto results = context.getResults(MessageSize { 16, 1 });
-    results.setDownSocket(kj::heap<CapnpToKjWebSocketAdapter>(
-        kj::addRef(*state), webSocket, kj::mv(shorteningPaf.promise)));
+    auto downPaf = kj::newPromiseAndFulfiller<kj::Promise<void>>();
+    auto downSocket = kj::heap<CapnpToKjWebSocketAdapter>(
+        kj::addRef(*state), webSocket, kj::mv(shorteningPaf.promise), kj::mv(downPaf.fulfiller));
+    results.setDownSocket(kj::mv(downSocket));
+
+    responsePumpTask = kj::joinPromises(kj::arr(kj::mv(upPumpTask), kj::mv(downPaf.promise)));
```

Comment:
Do we want `joinPromises()`, or `joinPromisesFailFast()`? If the former, it's probably worth a comment why, since we've found the latter's eager exception propagation more intuitive.